torchgeo/datasets/mdas.py

```diff
@@ -0,0 +1,347 @@
+
+    def _load_image(self, path: Path) -> Tensor:
+        """Load an image from a given path."""
+        with rio.open(path) as src:
+            img = src.read()
+            if img.dtype == np.uint16:
+                img = img.astype(np.int32)
+            if 'osm_landuse' in str(path):
+                img = np.vectorize(self.landuse_mapping.get)(img)
+
+            return torch.from_numpy(img)
+
+    def __getitem__(self, idx: int) -> dict[str, Any]:
+        """Return the dataset sample at the given index."""
+        sample_files = self.files[idx]
+        sample: dict[str, Any] = {}
+        for modality, path in sample_files.items():
+            if 'osm' in modality:
+                sample[f'{modality}_mask'] = self._load_image(path).long()
+            else:
+                sample[f'{modality}_image'] = self._load_image(path)
+
+        if self.transforms:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset."""
+        # check if each desired modality file exists in specified subarea
+        exists = []
+        for subarea in self.subareas:
+            for modality in self.modalities:
+                path = os.path.join(
+                    self.root,
+                    self.ds_root_name,
+                    subarea,
+                    f'{modality}_{self._format_subarea(subarea)}.tif',
+                )
+                if not os.path.exists(path):
+                    exists.append(False)
+                else:
+                    exists.append(True)
+        if all(exists):
+            return
+
+        # check if zip file downloaded
+        if os.path.exists(os.path.join(self.root, self.zipfilename)):
+            self._extract()
+            return
+
+        if not self.download:
+            raise DatasetNotFoundError(self)
+
+        self._download()
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+        extract_archive(os.path.join(self.root, self.zipfilename), self.root)
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        download_and_extract_archive(
+            self.url,
+            self.root,
+            filename=self.zipfilename,
+            md5=self.md5 if self.checksum else None,
+        )
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: str | None = None,
+    ) -> Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: A sample returned by `__getitem__`.
+            show_titles: Whether to display titles on the subplots.
+            suptitle: An optional super title for the plot.
+        """
+        ncols = len(sample)
+        fig, axs = plt.subplots(1, ncols, figsize=(5 * ncols, 5))
+
+        if ncols == 1:
+            axs = [axs]
+
+        for idx, (key, data) in enumerate(sample.items()):
```

Comment:
I wonder if we should use `self.modalities` to keep the order the same? Don't have a strong opinion here.